### DISCUSSION TITLE
Is it possible to use MSBuildWorkspace on a device without MSBuild.exe?

### DISCUSSION THREAD

**Discussion by KUNGERMOoN** [dotnet/roslyn#70325] [Q&A]
EDIT: Seems like [Microsoft.Build.Runtime](https://github.com/dotnet/msbuild/blob/main/src/MSBuild/README.md) might be what I'm looking for. I can't seem to find any documentation for it though.

Basically title.
An example use case would be using MSBuildWorkspace inside android app running on Xamarin or .Net MAUI - which don't have MSBuild installed and thus it wouldn't be possible to use something like `MSBuildLocator.RegisterDefaults` like as shown [here](https://gist.github.com/DustinCampbell/32cd69d04ea1c08a16ae5c4cd21dd3a3#setting-up-msbuild)

**Comment by jasonmalinowski** [comment]
Generally no, it's not supported. Resolving things like references relies on things like NuGet or the build tasks themselves, so without that you wouldn't get a terribly usable result back.
